src/target/riscv/riscv-013.c

```diff
@@ -2427,7 +2427,8 @@ static int write_memory_abstract(struct target *target, target_addr_t address,
 	bool updateaddr = true;
 	for (uint32_t c = 0; c < count; c++) {
 		/* Move data to arg0 */
-		riscv_reg_t value = *p;
+		riscv_reg_t value = 0;
+		memcpy(&value, p, size);
```

Comment:
That probably doesn't work on big-endian hosts. Can you do something like https://github.com/riscv/riscv-openocd/blob/riscv/src/target/riscv/riscv-013.c#L2764? (For bonus points, share the code.)

Reply:
Thanks for the feedback. Much cleaner and more portable now. I did a little digging and found `write_to_buf()`, which is already being used when reading from memory. So I used that for `read_memory_abstract()` replacing that memcpy. Then I wrote `read_from_buf()` and used it in `write_memory_abstract()` for its memcpy.

I didn't, however, replace that other replicated code. I just dropped a TODO in the three spots that can use it. I can't really test it on our platform. ...though I *just* realized that I might be able to use Spike. I'm happy to give that a shot if like.